src/auth/src/credentials/user.rs

```diff
@@ -135,17 +136,13 @@ where
 
     async fn headers(&self) -> Result<Vec<(HeaderName, HeaderValue)>> {
         let token = self.token().await?;
-        let mut value = HeaderValue::from_str(&format!("{} {}", token.token_type, token.token))
-            .map_err(errors::non_retryable)?;
-        value.set_sensitive(true);
-        let mut headers = vec![(AUTHORIZATION, value)];
-        if let Some(project) = &self.quota_project_id {
-            headers.push((
-                HeaderName::from_static(QUOTA_PROJECT_KEY),
-                HeaderValue::from_str(project).map_err(errors::non_retryable)?,
-            ));
-        }
-        Ok(headers)
+        build_headers(
+            &token,
+            &self.quota_project_id,
+            AUTHORIZATION,
+            |token| HeaderValue::from_str(&format!("{} {}", token.token_type, token.token))
```

Comment:
Can we add a method in header util that builds the Bearer value and pass in that method here? This same logic is repeated thrice for now and will be done more times in future.

Reply:
@sai-sunder-s: Yes, I think this can be refactored further. Let me give it a try.